praw/models/reddit/emoji.py

```diff
@@ -0,0 +1,71 @@
+"""Provide the EMOJI class."""
+from ...const import API_PATH
+from .base import RedditBase
+
+
+class Emoji(RedditBase):
+    """An individual Emoji object."""
+
+    __hash__ = RedditBase.__hash__
+    STR_FIELD = 'name'
+
+    def __init__(self, reddit, subreddit, name, _data=None):
+        """Construct an instance of the Emoji object."""
+        self.reddit = reddit
+        self.subreddit = subreddit
+        self.name = name
+        super(Emoji, self).__init__(reddit, _data)
+
+    def add(self, filepath):
+        """Add an emoji to this subreddit.
+
+        :param emoji: An emoji name (e.g., ``'cake'``) or
+            :class:`~.Emoji` instance.
+
+        To add ``'cake'`` to the subreddit ``'praw_test'`` try:
+
+        .. code:: python
+
+           reddit.subreddit('praw_test').emoji['cake'].add('cake.png')
+
+        """
+        data = None
+        if filepath.lower().endswith('.png'):
+            data = {'filepath': filepath, 'mimetype': 'image/png'}
+        elif filepath.lower().endswith('.jpg') \
+                or filepath.lower().endswith('.jpeg'):
+            data = {'filepath': filepath, 'mimetype': 'image/jpeg'}
+        if data is not None:
+            url = API_PATH['emoji_lease'].format(
+                subreddit=self.subreddit, method='add')
+            s3_lease = self._reddit.post(url, data=data)['s3UploadLease']
```

Comment:
I'm a little concerned about what might happen if this request fails. E.g., what if the filepath doesn't have an extension. This isn't something we need to address right now, but maybe add a comment like `# until we learn otherwise, assume this request always succeeds`. Having such a comment makes it clear we've at least thought about the possibility of it failing.